tasks/human_eval_129.rs

```diff
@@ -9,10 +9,450 @@ use vstd::prelude::*;
 
 verus! {
 
-// TODO: Put your solution (the specification, implementation, and proof) to the task here
+// specification
+pub closed spec fn path_less_than(path1: Seq<int>, path2: Seq<int>) -> bool
+    recommends
+        path1.len() == path2.len(),
+{
+    (exists|n: int|
+        (0 <= n < path1.len() && path1[n] < path2[n] && (forall|i: int|
+            0 <= i < n ==> path1[i] == path2[i])))
+}
+
+pub closed spec fn is_adjacent<const N: usize>(
+    grid: Seq<Seq<int>>,
+    m: int,
+    n: int,
+    coords: (int, int),
+) -> bool
+    recommends
+        grid.len() == N,
+        forall|i: int| (0 <= i < N ==> #[trigger] grid[i].len() == N),
+{
+    &&& (0 <= coords.0 <= N - 1)
+    &&& (0 <= coords.1 <= N - 1)
+    &&& (grid[coords.0][coords.1]) == m
+    &&& ((coords.0 < N - 1 && grid[coords.0 + 1][coords.1] == n) || (coords.0 > 0
```

Comment:
You might want to add a `#[verusfmt::skip]`tag to this bit, so that the lines line up better, so the regular structure is more apparent.

Reply:
done!